### PR TITLE
Facebook Authentication

### LINKED ISSUE #683: Add "Log in with Facebook" for new accounts
(later following up with twitter/other social media)

OmniAuth is a great starting point: https://github.com/omniauth/omniauth

Let's begin with Facebook!

See these "strategies" to support FB, Twitter, Google, etc:

https://github.com/omniauth/omniauth/wiki/List-of-Strategies

****

@steviepubliclab - starting this issue here :)

### PR DESCRIPTION
**Pull request by SidharthBansal**
Fixes #683 
To Dos
* [ ] User_session_controller update
* [x] Views update
* [ ] Write tests for the simple login through fb(without integration)
* [ ] Integration with the current model
* [ ] Write tests
* [ ] Write Documentation for the fiagro gem
* [ ] Write Documentation for omniauth 
* [ ] Write Documentaion for the omniauth-facebook

### CODE REVIEW

**Comment by biswesh456** [comment]
Interesting. For me in the method create_with_omniauth(auth) caused problem related to password validation hence I had to include user.password_confirmation in it. Does it not give you any error @SidharthBansal ? ( Just curious !!!)

**Comment by SidharthBansal** [comment]
I am currently working on it. A lot of things are pending in this pr

**Comment by jywarren** [comment]
Wow, this is a big one. If there's any way to implement some of these systems in smaller pieces, I'm open to that -- but I understand if it seems impractical. I guess, i'm wondering also if we could implement it only offer login by facebook when a URL flag is set like `?fb=true` (checking `params[:fb]` in the view) so that we can test this out for a bit before presenting it to everyone on the live site by default. Make sense? 

Good luck with this!!!

File: app/views/layouts/_header.html.erb [hunk 1]
```diff
@@ -177,6 +177,7 @@
                     <a class="btn btn-default" href="/signup" tabindex="4"><%= t('layout._header.login.sign_up') %></a>
                   </p>
 
+                  <p><a href="/auth/facebook">Login with Facebook</a></p>
```
Comment (jywarren) [inline app/views/layouts/_header.html.erb]: Here -- we could do `<% if params[:fb] %> ... <% end %>` to only show this when that flag is set, you know?

Comment (SidharthBansal) [reply app/views/layouts/_header.html.erb]: Yeah!!!   I also want to test this feature on production before launching to public. Thanks for the hint. I will try to do it.

**Comment by SidharthBansal** [comment]
We can write the documentations in separate prs. But all other things needs to be done in a single PR.
Thanks

**Comment by jywarren** [comment]
OK, great -- just tell me how I can help keep these organized and if you get stuck!

**Comment by jywarren** [comment]
Hi @SidharthBansal how's this coming along? is it worth trying to see if we could merge in functional subsections of this project in smaller parts? Like adding gems, etc etc, adding back-end infrastructure... what do you think?

**Comment by SidharthBansal** [comment]
We can do that. But that will not result in functionality until all commits/ prs are merged. 
Ok. I am closing this now and pushing subsections one by one.
Thanks

**Pull request closed without merging**